kubetest/aksengine.go

```diff
@@ -214,7 +214,11 @@ func (c *aksEngineDeployer) SetCustomCloudProfileEnvironment() error {
 			Timeout: 30 * time.Second,
 		}
 		endpointsresp, err := httpClient.Get(metadataURL)
-		if err != nil || endpointsresp.StatusCode != 200 {
+		if err != nil {
```

Comment:
I don't think this matches the intent of what was here before.  Is there a test that demonstrates the behavior this is simplifying?

https://golang.org/pkg/net/http/#Client.Get

> A non-2xx response doesn't cause an error.